tools/config-gen/params.yaml

```diff
@@ -39,25 +347,117 @@
+  config-path: "file-cache.max-size-mb"
+  type: "int"
+  usage: "Maximum size of the file-cache in MiBs"
+  default: "-1"
+
+- flag-name: "cache-file-for-range-read"
+  config-path: "file-cache.cache-file-for-range-read"
+  type: "bool"
+  usage: "Whether to cache file for range reads."
+
+- flag-name: "enable-crc-check"
+  config-path: "file-cache.enable-crc-check"
+  type: "bool"
+  usage: "Performs CRC check to ensure that file is correctly downloaded into cache."
+  default: true
+
+- flag-name: "enable-parallel-downloads"
+  config-path: "file-cache.enable-parallel-downloads"
+  type: "bool"
+  usage: "Enable parallel downloads."
+
+- flag-name: "download-parallelism-per-file"
+  config-path: "file-cache.download-parallelism-per-file"
+  type: "int"
+  usage: "Number of concurrent download requests per file."
+  default: "10"
+
+- flag-name: "max-download-parallelism"
+  config-path: "file-cache.max-download-parallelism"
+  type: "int"
+  usage: "Sets an uber limit of number of concurrent file download requests that are made across all files."
+  default: "-1"
+
+- flag-name: "read-request-size-mb"
+  config-path: "file-cache.read-request-size-mb"
+  type: "int"
+  usage: "Size of chunks in MiB that each concurrent request downloads."
+
+- flag-name: "cache-dir"
+  config-path: "cache-dir"
+  type: "resolvedPath"
+  usage: "Enables file-caching. Specifies the directory to use for file-cache."
+
+- flag-name: "enable-empty-managed-folders"
+  config-path: "list.enable-empty-managed-folders"
+  type: "bool"
   usage: >-
-    The protocol used for communicating with the GCS backend.
-    Value can be 'http1' (HTTP/1.1), 'http2' (HTTP/2) or 'grpc'.
+    This handles the corner case in listing managed folders.
+    There are two corner cases (a) empty managed folder (b) nested managed folder which doesn't contain any descendent as object.
+    This flag always works in conjunction with --implicit-dirs flag.
+    (a) If only ImplicitDirectories is true, all managed folders are listed other than above two mentioned cases.
+    (b) If both ImplicitDirectories and EnableEmptyManagedFolders are true, then all the managed folders are listed including the above-mentioned corner case.
+    (c) If ImplicitDirectories is false then no managed folders are listed irrespective of enable-empty-managed-folders flag.
+
+- flag-name: "grpc-conn-pool-size"
+  config-path: "gcs-connection.grpc-conn-pool-size"
+  type: "int"
+  usage: "The number of gRPC channel in grpc client."
+  deprecated: true
+  deprecation-warning: "Experimental flag: can be removed in a minor release."
+
+- flag-name: "enable-hns"
+  config-path: "enable-hns"
+  type: "bool"
+  usage: "Enables support for HNS buckets"
+
+- flag-name: "log-rotate-max-log-file-size-mb"
+  config-path: "logging.log-rotate.max-file-size-mb"
+  type: "int"
+  default: "512"
+  usage: "the maximum size in megabytes that a log file can reach before it is rotated."
+
+- flag-name: "log-rotate-backup-file-count"
```

Comment:
NIT: can we keep all related flags together?

Reply:
We can do that at a later stage. For now, I've tried to keep the order like it was in flags.go so that I don't miss any entries.